### PR TITLE
Add support for taints and kube extra args

### PR DESCRIPTION
**Pull request by vaijab**

### CODE REVIEW

File: pkg/constants/constants.go [hunk 1]
```diff
@@ -6,7 +6,7 @@ const (
 	// DefaultNetworkProvider specifies what CNI provider to install
 	DefaultNetworkProvider = "canal"
 	// DefaultKetoK8Image specifies the image to use for keto-k8 container
-	DefaultKetoK8Image = "quay.io/ukhomeofficedigital/keto-k8:v0.1.1-b1"
+	DefaultKetoK8Image = "quay.io/ukhomeofficedigital/keto-k8:v0.2.2-extraargs"
```
Comment (lewismarshall) [inline pkg/constants/constants.go]: Awsome - can use v0.2.2 now

**Comment by vaijab** [comment]
> Ran out of time to test

No worries, can leave this for early Monday.

**Comment by KashifSaadat** [comment]
Gave this a test with the new keto-k8 image. When initially building the cluster, the API doesn't become available on the master nodes:

keto-k8 service:
```
level=info msg="Running:kubeadm alpha phase kubeconfig client-certs --client-name kubernetes-admin --server https://keto-kashtst01-elb-X.eu-west-1.elb.amazonaws.com --organization system:masters"
level=info msg="Saving:"/etc/kubernetes/admin.conf""
level=info msg="Running:kubeadm alpha phase kubeconfig client-certs --client-name system:node:ip-172-30-2-72.eu-west-1.compute.internal --server https://keto-kashtst01-elb-X.eu-west-1.elb.amazonaws.com --organization system:nodes"
level=info msg="Saving:"/etc/kubernetes/kubelet.conf""
level=info msg="Running:kubeadm alpha phase kubeconfig client-certs --client-name system:kube-controller-manager --server https://keto-kashtst01-elb-X.eu-west-1.elb.amazonaws.com"
level=info msg="Saving:"/etc/kubernetes/controller-manager.conf""
level=info msg="Running:kubeadm alpha phase kubeconfig client-certs --client-name system:kube-scheduler --server https://keto-kashtst01-elb-X.eu-west-1.elb.amazonaws.com"
level=info msg="Saving:"/etc/kubernetes/scheduler.conf""
[apiclient] Created API client, waiting for the control plane to become ready
```

kubelet service:
```
W0724 09:31:07.274174    2692 cni.go:189] Unable to update cni config: No networks found in /etc/cni/net.d
E0724 09:31:07.274650    2692 kubelet.go:2136] Container runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:docker: network plugin is not ready: cni config uninitialized
E0724 09:31:07.628099    2692 reflector.go:190] k8s.io/kubernetes/pkg/kubelet/kubelet.go:408: Failed to list *v1.Node: Get https://keto-kashtst01-elb-X.eu-west-1.elb.amazonaws.com/api/v1/nodes?fieldSelector=metadata.name%3Dip-172-30-2-72.eu-west-1.compute.internal&resourceVersion=0: EOF
E0724 09:31:07.724438    2692 reflector.go:190] k8s.io/kubernetes/pkg/kubelet/config/apiserver.go:46: Failed to list *v1.Pod: Get https://keto-kashtst01-elb-X.eu-west-1.elb.amazonaws.com/api/v1/pods?fieldSelector=spec.nodeName%3Dip-172-30-2-72.eu-west-1.compute.internal&resourceVersion=0: EOF
E0724 09:31:08.198338    2692 reflector.go:190] k8s.io/kubernetes/pkg/kubelet/kubelet.go:400: Failed to list *v1.Service: Get https://keto-kashtst01-elb-X.eu-west-1.elb.amazonaws.com/api/v1/services?resourceVersion=0: EOF
E0724 09:31:08.632268    2692 reflector.go:190] k8s.io/kubernetes/pkg/kubelet/kubelet.go:408: Failed to list *v1.Node: Get https://keto-kashtst01-elb-X.eu-west-1.elb.amazonaws.com/api/v1/nodes?fieldSelector=metadata.name%3Dip-172-30-2-72.eu-west-1.compute.internal&resourceVersion=0: EOF
E0724 09:31:08.729478    2692 reflector.go:190] k8s.io/kubernetes/pkg/kubelet/config/apiserver.go:46: Failed to list *v1.Pod: Get https://keto-kashtst01-elb-X.eu-west-1.elb.amazonaws.com/api/v1/pods?fieldSelector=spec.nodeName%3Dip-172-30-2-72.eu-west-1.compute.internal&resourceVersion=0: EOF
E0724 09:31:09.002958    2692 event.go:209] Unable to write event: 'Patch https://keto-kashtst01-elb-X.eu-west-1.elb.amazonaws.com/api/v1/namespaces/default/events/ip-172-30-2-72.eu-west-1.compute.internal.14d4394504688cfe: EOF' (may retry after sleeping)
E0724 09:31:09.201996    2692 reflector.go:190] k8s.io/kubernetes/pkg/kubelet/kubelet.go:400: Failed to list *v1.Service: Get https://keto-kashtst01-elb-X.eu-west-1.elb.amazonaws.com/api/v1/services?resourceVersion=0: EOF
E0724 09:31:09.383611    2692 eviction_manager.go:238] eviction manager: unexpected err: failed GetNode: node 'ip-172-30-2-72.eu-west-1.compute.internal' not found
```

**Comment by KashifSaadat** [comment]
Labels and taint KVs don't appear to be passed correctly on new compute pool creation. When inspecting the kubelet service config, there's no reference to the taint args specified or even labels (which used to work):
```
--node-labels= \
```

**Comment by KashifSaadat** [comment]
Debugging:
- Keto updates cloudformation template accordingly with Labels & Taints, this is present and viewable
- Keto-K8 is referencing the correct Labels and Taints keys
- Keto-K8 attempts to retrieve Labels & Taints via use of `GetNodeData` on the `NodeData` model, which returns an empty map

I believe the issue is around the following lines, not confirmed yet: https://github.com/UKHomeOffice/keto/blob/master/pkg/cloudprovider/providers/aws/node.go#L43-L45

**Pull request merged**